### ISSUE TITLE
funannotate update with mysql

### ISSUE THREAD

**Issue by Nitin123-4** [nextgenusfs/funannotate#998]
Hi Team,

I did the required settings for funannotate to use mysql i.e. https://github.com/PASApipeline/PASApipeline/wiki/setting-up-pasa-mysql.

It is running fine with 12 cpus.

funannotate update -i Genome_funannotate_train/ --cpus 12 --pasa_db mysql

Is there any estimate how long it will take?

Previous annotation consists of: 28,846 protein coding gene models and 683 non-coding gene models,

**Comment by Nitin123-4** [comment]
Hi Team, Any idea about this ?

**Comment by nextgenusfs** [comment]
I guess it depends on how many transcripts you had in the initial round. It should be faster than when it ran `train`.  I'm assuming that you ran train with the same mysql setup. The update step runs PASA genome comparison, so it takes your gene models from `predict` and looks at the alignments in the PASA database -- it can predict some new genes and/or alter some coding sequences.  It runs this step two times iteratively. If it is still running after 7 days then I'd think something is wrong.  I would guess several hours would be more typical.

**Comment by Nitin123-4** [comment]
Hi Thanks for your reply. I ran funannotate train with sqlite. I am running update with mysql.

**Comment by nextgenusfs** [comment]
That won't work.

**Comment by Nitin123-4** [comment]
Oh Okay. So funannotate train should also be with mysql and update also then it should be faster?

**Comment by nextgenusfs** [comment]
Yeah it tries to reuse the database. You could just run a fresh update but need to pass all the reads, etc and direct to new output folder. Update is capable of running the data from scratch as well.
